src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/DelegatingAIFunctionDefinition.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Microsoft.Shared.Diagnostics;
+
+#pragma warning disable SA1202 // Elements should be ordered by access
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides an optional base class for an <see cref="AIFunctionDefinition"/> that passes through calls to another instance.
+/// </summary>
+public class DelegatingAIFunctionDefinition : AIFunctionDefinition
```

Comment:
I'm not clear on the use case for this, except to support `AsDefinitionOnly`. Could it be `internal`?

Reply:
Yes, I had it public for completeness, but making it internal, at least for now, makes sense. It's much more applicable at the AIFunction level where it's more common to want to augment invocation behavior.